code/modules/transport/tram/tram_floors.dm

```diff
@@ -0,0 +1,243 @@
+/turf/open/floor/noslip/tram_plate/energized/proc/toast(mob/living/future_tram_victim)
+	var/datum/transport_controller/linear/tram/tram = find_tram()
+
+	// Check for stopped states.
+	if(!tram || !tram.controller_operational || !inbound || !outbound)
+		return FALSE
+
+	var/obj/structure/transport/linear/tram/tram_part = tram.return_closest_platform_to(src)
+
+	if(QDELETED(tram_part))
+		return FALSE
+
+	// Everything will be based on position and travel direction
+	var/plate_pos
+	var/tram_pos
+	var/tram_velocity_sign // 1 for positive axis movement, -1 for negative
+	// Try to be agnostic about N-S vs E-W movement
+	if(tram.travel_direction & (NORTH|SOUTH))
+		plate_pos = y
+		tram_pos = tram_part.y
+		tram_velocity_sign = tram.travel_direction & NORTH ? 1 : -1
+	else
+		plate_pos = x
+		tram_pos = tram_part.x
+		tram_velocity_sign = tram.travel_direction & EAST ? 1 : -1
+
+	// How far away are we? negative if already passed.
+	var/approach_distance = tram_velocity_sign * (plate_pos - (tram_pos + (DEFAULT_TRAM_LENGTH * 0.5)))
+
+	// Check if our victim is in the active path of the tram.
+	if(!tram.controller_active)
+		return FALSE
+	if(approach_distance < 0)
+		return FALSE
+	playsound(src, SFX_SPARKS, 75, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	audible_message(
+		span_hear("You hear an electric crackle when you step on the plate...")
+	)
+	if(tram.travel_direction & WEST && inbound < tram.idle_platform.platform_code)
+		return FALSE
+	if(tram.travel_direction & EAST && outbound > tram.idle_platform.platform_code)
+		return FALSE
+	if(approach_distance >= AMBER_THRESHOLD_NORMAL)
+		return FALSE
+
+	// Finally the interesting part where they ACTUALLY get hit!
+	notify_ghosts("[future_tram_victim] has fallen in the path of an oncoming tram!", source = future_tram_victim, action = NOTIFY_ORBIT, header = "Electrifying!")
+	future_tram_victim.electrocute_act(15, src, 1)
+	return TRUE
+
+/turf/open/floor/glass/reinforced/tram/Initialize(mapload)
+	. = ..()
+	RemoveElement(/datum/element/atmos_sensitive, mapload)
+
+/turf/open/floor/glass/reinforced/tram
+	name = "tram bridge"
+	desc = "It shakes a bit when you step, but lets you cross between sides quickly!"
+
+/obj/structure/thermoplastic
+	name = "tram"
+	desc = "A lightweight thermoplastic flooring."
+	icon = 'icons/turf/floors.dmi'
+	icon_state = "tram_dark"
+	density = FALSE
+	anchored = TRUE
+	armor_type = /datum/armor/tram_structure
+	max_integrity = 750
+	layer = TRAM_FLOOR_LAYER
+	plane = FLOOR_PLANE
+	obj_flags = BLOCK_Z_OUT_DOWN | BLOCK_Z_OUT_UP
+	appearance_flags = PIXEL_SCALE|KEEP_TOGETHER
+	var/secured = TRUE
+	var/floor_tile = /obj/item/stack/thermoplastic
+
+/obj/structure/thermoplastic/light
+	icon_state = "tram_light"
+	floor_tile = /obj/item/stack/thermoplastic/light
+
+/obj/structure/thermoplastic/examine(mob/user)
+	. = ..()
+
+	if(secured)
+		. += span_notice("It is secured with a set of <b>screws.</b>")
+	else
+		. += span_notice("You can <b>crowbar</b> to remove the tile.")
```

Comment:
there's a define you could be using here instead, i think EXAMINE_HINT or something? for the bold.